### ISSUE TITLE
ENH: 

### ISSUE THREAD

**Issue by franzhaas** [pandas-dev/pandas#44049]
#### Is your feature request related to a problem?

When using jupyter-book with the pdflatex i can create beautiful tables from data frames, but when using a styler object this does not work at all.

I suspect this is due to differences in how to use the _repr_latex_ methods for styler and for dataframe objects.

#### Describe the solution you'd like

I would like to be able to use styler object like dataframe objects to show up in the pdflatex generated pdf.

I was able to achieve that by using this function.:

````python
styler = fancy_styled_pandas_dataframe


def latex_save_styler(styler):
    def a(*args, **kwargs):
        return styler.data._repr_latex_(*args, **kwargs)
    styler._repr_latex_ = a
    return styler

latex_save_styler(styler)
````

see also this discussion.: https://github.com/executablebooks/jupyter-book/issues/1512

#### API breaking implications

I do believe the offending property are the mandatory keyword arguments for the _repr_latex methof of the styler object.

parse_index, sparse_columns

#### Describe alternatives you've considered

I am unsure if this is a problem of pandas, or expected behaviour. (and the way pandas is called from jupyter-book should be changed) however, the given code solved my problem at hand, it may or may not be usefull for other people or in understanding/fixing the issue.

#### Additional context

I am using pandas ==1.3.3 and jupyter-book==0.11.3

**Comment by franzhaas** [comment]
I am sorry, i do not get it.

````python
def latex_save_styler(styler):
    def a(*args, **kwargs):
        return styler._render_latex(sparse_index=True, sparse_columns=True)
    styler._repr_latex_ = a
    return styler
latex_save_styler(s)
````

works for me in the sense as it creates latex with a table.

So it apears to me that jupyter-book is calling (directly or indirectly) the _repr_latex_ method of the styler, but not with those mandatory parameters.

I am confused about this code block.:

````python
def test_repr_option(styler):
    assert "<style" in styler._repr_html_()[:6]
    assert styler._repr_latex_() is None
    with option_context("styler.render.repr", "latex"):
        assert "\\begin{tabular}" in styler._repr_latex_()[:15]
        assert styler._repr_html_() is None
````
I would have expected this not work as styler._repr_latex_() has 2 mandatory keyword arguments, at least on my setup (pandas 1.3.3).

Is there an interface change on the way that i missed? Or is my setup wired? If the test_repr_option given a styler object passes, i would expect it to not have the problem i am experiencing.

**Comment by attack68** [comment]
1.3.3 does not have that test, nor the functionality. The PR link I gave is for master, unreleased version 1.4.0.

With your `latex_save_styler` I think you will have the result that in Jupyter Notebook in a browser `latex_save_styler(s)` will render as an HTML table since you have not overwritten the `repr_html` method and that will take precedence, where it still exists, but when using a converter it will generate a LaTeX table since you have created the `repr_latex` method which the converter will use directly.

**Comment by mroeschke** [comment]
Appears that this feature will be available in 1.4 which is slated to be released in December. In the meantime, the feature is available if the library is built from source. Closing for now, but can reopen if there is a bug in the functionality
